jerry-core/ecma/builtin-objects/ecma-builtin-string-prototype.cpp

```diff
@@ -344,8 +347,245 @@ static ecma_completion_value_t
 ecma_builtin_string_prototype_object_match (ecma_value_t this_arg, /**< this argument */
                                             ecma_value_t arg) /**< routine's argument */
 {
-  ECMA_BUILTIN_CP_UNIMPLEMENTED (this_arg, arg);
+  ecma_completion_value_t ret_value = ecma_make_empty_completion_value ();
+
+  /* 1. */
+  ECMA_TRY_CATCH (this_check_coercible_value,
+                  ecma_op_check_object_coercible (this_arg),
+                  ret_value);
+
+  /* 2. */
+  ECMA_TRY_CATCH (this_to_string_value,
+                  ecma_op_to_string (this_arg),
+                  ret_value);
+
+  ecma_value_t regexp_value = 0;
```

Comment:
`0` is not a valid value for `ecma_value_t`.

Could you, please, change this to `ecma_make_simple_value (ECMA_SIMPLE_VALUE_EMPTY)` and check below to `!ecma_is_value_empty`?

Also, maybe, the initialization and check could be removed, as compiler checks for accessing uninitialized values.